soc/arm/nuvoton_npcx/common/power.c

```diff
@@ -84,6 +86,38 @@ enum {
 	NPCX_STANDARD_WAKE_UP,
 };
 
+/**
+ * Collect all io pads which enable 'low-voltage-input' from 'pinctrl' node
+ * This list is for bypass functions which solve power consumption issue in
+ * NPCX7 series. NPCX9 series have fixed it.
+ */
+#define NPCX_IO_PADS_LVOL_ITEMS(node_id)				\
+	COND_CODE_1(DT_PROP(node_id, low_voltage_input),		\
+		(Z_PINCTRL_NPCX_IO_PAD_CONF_INIT(node_id, io_pad)), ())
+
+#define NPCX_IO_PADS_LVOL_LIST						\
+	{DT_FOREACH_CHILD(DT_NODELABEL(pinctrl), NPCX_IO_PADS_LVOL_ITEMS)}
+
+const pinctrl_soc_pin_t lvol_pads[] = NPCX_IO_PADS_LVOL_LIST;
+
+static void  npcx_power_lvol_restore_io_pads(void)
+{
+	for (int i = 0; i < ARRAY_SIZE(lvol_pads); i++) {
+		npcx_gpio_enable_io_pads(
+			npcx_get_gpio_dev(lvol_pads[i].cfg.io_pad.port),
+			lvol_pads[i].cfg.io_pad.bit);
+	}
+}
+
+static void npcx_power_lvol_suspend_io_pads(void)
+{
+	for (int i = 0; i < ARRAY_SIZE(lvol_pads); i++) {
+		npcx_gpio_disable_io_pads(
+			npcx_get_gpio_dev(lvol_pads[i].cfg.io_pad.port),
+			lvol_pads[i].cfg.io_pad.bit);
+	}
+}
```

Comment:
why is this done at `soc/` level? why don't you have `default`/`sleep` states to handle low power configurations?

Reply:
This is a workaround method only for npcx7 series to resolve the power consumption issue. (This method is deprecated in npcx9 and the following series.) Since it will cut all connections of all io pads which enables 1.8 V detection level, our hardware engineer suggests restoring the connections after ec wake-up ASAP in case of missing events of IO pads. There is still a latency and risk if we put them in the hook functions of related power states.